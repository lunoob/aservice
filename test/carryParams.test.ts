/**
* @fileoverview 测试携带额外参数拦截器（请求拦截器）
* @author Luoob
*/

import { createRequest, createReqInterceptor, carryParamsInterceptor } from '../esm'

describe('测试携带参数功能', () => {
    test('携带参数是否正确', () => {
        let params: any = {}

        createReqInterceptor(carryParamsInterceptor({
            name: 'luoob',
            age: '18'
        }))

        createReqInterceptor(carryParamsInterceptor(() => {
            return {
                dynamic: 1
            }
        }))

        createReqInterceptor((config) => {
            params = config.params
            return config
        })

        const http = createRequest({ baseURL: '/' })

        return http.request('get')('/1')({}).then(() => {
            expect(params).toMatchSnapshot()
        })
    })
})
