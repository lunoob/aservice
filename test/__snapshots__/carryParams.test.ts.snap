// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试携带参数功能 携带参数是否正确 1`] = `
Object {
  "age": "18",
  "dynamic": 1,
  "name": "luoob",
}
`;
